### PR TITLE
Ark.Tools.HealthCheck Extension

### PR DESCRIPTION
**Pull request by MichaelGeraghty**

### CODE REVIEW

File: Ark.Tools.AspNetCore.HealthChecks/Ark.Tools.AspNetCore.HealthChecks.csproj [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	<ItemGroup>
+		<FrameworkReference Include="Microsoft.AspNetCore.App" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<PackageReference Include="AspNetCore.HealthChecks.AzureServiceBus" Version="3.2.2" />
+		<PackageReference Include="AspNetCore.HealthChecks.Network" Version="3.1.3" />
+		<PackageReference Include="AspNetCore.HealthChecks.Oracle" Version="3.1.2" />
+		<PackageReference Include="AspNetCore.HealthChecks.Publisher.ApplicationInsights" Version="3.1.1" />
+		<PackageReference Include="AspNetCore.HealthChecks.RavenDB" Version="3.1.1" />
```
Comment (AndreaCuneo) [inline Ark.Tools.AspNetCore.HealthChecks/Ark.Tools.AspNetCore.HealthChecks.csproj:35]: Exclude RavenDB dependency from base package, we don't use it often. can be added on Application side

File: Ark.Tools.AspNetCore.HealthChecks/HealthCheckDto.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ark.Tools.AspNetCore.HealthChecks
+{
+    public class HealthCheckDto
```
Comment (AndreaCuneo) [inline Ark.Tools.AspNetCore.HealthChecks/HealthCheckDto.cs:9]: unused?

File: Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (AndreaCuneo) [inline Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj:1]: duplicated file?

File: WebApplicationDemo/Startup.cs [hunk 4]
```diff
@@ -70,6 +71,12 @@ public override void ConfigureServices(IServiceCollection services)
 			})
 			;
 
+			//HealthChecks
+			services.AddArkHealthChecks()
```
Comment (AndreaCuneo) [inline WebApplicationDemo/Startup.cs:75]: I think AddArkHealthChecks() should be done in the common Startups.
The application should only Add more checks, but UI and Endpoint should be there by default.

File: WebApplicationDemo/Startup.cs [hunk 5]
```diff
@@ -101,6 +108,9 @@ public override void ConfigureServices(IServiceCollection services)
 		public override void Configure(IApplicationBuilder app)
 		{
 			base.Configure(app);
+
+			//HealthChecks
+			app.UseArkHealthChecks();
```
Comment (AndreaCuneo) [inline WebApplicationDemo/Startup.cs:113]: should be in common startups

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj (hunk 3 above)
Comment (MichaelGeraghty) [reply Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj:1]: Yes or copied csproj for nuget configurations . Is there something incorrect?

File: Ark.Tools.AspNetCore.HealthChecks/Ark.Tools.AspNetCore.HealthChecks.csproj (hunk 1 above)
Comment (MichaelGeraghty) [reply Ark.Tools.AspNetCore.HealthChecks/Ark.Tools.AspNetCore.HealthChecks.csproj:35]: Remove the reference from AspNetCore?

Comment (AndreaCuneo) [reply Ark.Tools.AspNetCore.HealthChecks/Ark.Tools.AspNetCore.HealthChecks.csproj:35]: no, just to RavenDB

File: Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj (hunk 3 above)
Comment (AndreaCuneo) [reply Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj:1]: this looks duplicated, there is both 
Ark.Tools.AspNetCore.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj
Ark.Tools.HealthCheck/Ark.Tools.AspNetCore.HealthCheck.csproj

the latter has project and folder that doesn't match.

File: Ark.Tools.Nodatime.SystemTextJson/Ark.Tools.Nodatime.SystemTextJson.csproj [hunk 6]
```diff
@@ -11,9 +11,7 @@
     <PackageReleaseNotes></PackageReleaseNotes>
 
     <Version>$(ArkToolsCoreVersion)</Version>
-    <!--<VersionSuffix>$(ArkToolsCoreVersionSuffix)</VersionSuffix>-->
```
Comment (MichaelGeraghty) [inline Ark.Tools.Nodatime.SystemTextJson/Ark.Tools.Nodatime.SystemTextJson.csproj:14]: To add beta versions to unavailability I need beta version of this .
Is there a need to remove suffix ability again in future?

Comment (MichaelGeraghty) [reply Ark.Tools.Nodatime.SystemTextJson/Ark.Tools.Nodatime.SystemTextJson.csproj:14]: Even with a beta-02 version listed I still have an error attempting to Install  Ark.Tools.AspNetCore 2.7.2 Beta-02
NU1102: Unable to find package Ark.Tools.Nodatime.SystemTextJson with version (>= 2.7.2)
  - Found 7 version(s) in nuget.org [ Nearest version: 2.7.2-beta02 ]

Comment (MichaelGeraghty) [reply Ark.Tools.Nodatime.SystemTextJson/Ark.Tools.Nodatime.SystemTextJson.csproj:14]: So AspNetCore has a dependency on 2.7.2
it seems the suffix is missing somewhere else

**Comment by MichaelGeraghty** [comment]
> rename this to 'Service' to make apparent is expected to be something that is already part of the Application.

Ok, is it correct now?

**Comment by AndreaCuneo** [comment]
> > rename this to 'Service' to make apparent is expected to be something that is already part of the Application.
> 
> Ok, is it correct now?

Not really. the IService should not return an HealthCheckResult.

The point is that if I want to 'implement' and HealthCheck using the HealthCheck library, I would use the HealthCheck approach, not the lambda.

The Lambda is for adding a quick check or some service, already present in the solution, maybe even calling a method existing that returns some application data, like a ReadContracts(page:0, size:1) and re-use that as a quick 'Check' in startup.

**Review by AndreaCuneo — APPROVED** [review]

**Pull request merged**